src/compiler/utilities.ts

```diff
@@ -647,7 +647,7 @@ namespace ts {
     }
 
     export function getLeadingCommentRangesOfNode(node: Node, sourceFileOfNode: SourceFile) {
-        return getLeadingCommentRanges(sourceFileOfNode.text, node.pos);
+        return node.kind !== SyntaxKind.JsxText ? getLeadingCommentRanges(sourceFileOfNode.text, node.pos) : undefined;

```

Comment:
`JsxText` nodes are children of `JsxElement`s that are not `JsxElements` themselves or curly-brace-delimited assignment expressions. In the following example, the underlined is `JsxText`:
```tsx
<div>Click me!<h1/></div>
     ~~~~~~~~~
```

We essentially treat them like string literals, though differences in the way they are delimited mean that `getLeadingCommentRanges` cannot handle them correctly without some information from the parser, which `getLeadingCommentRanges` has been designed to avoid.